Analysis/tasks.py

```diff
@@ -123,58 +134,81 @@ def create_branch_map(self):
         input_file = self.input()[0]
+        isTTCR,isDYCR,is2p5,isSC=getCustomisations(self.customisations,self.version)
         print(f'input file is {input_file.path}')
         global_config = os.path.join(self.ana_path(), 'config','HH_bbtautau', f'global.yaml')
         unc_config = os.path.join(self.ana_path(), 'config',self.period, f'weights.yaml')
         sample_type = self.samples[sample_name]['sampleType'] if sample_name != 'data' else 'data'
         HistProducerFile = os.path.join(self.ana_path(), 'Analysis', 'HistProducerFile.py')
         print(f'output file is {self.output().path}')
+        channels='eTau,muTau,tauTau'
+        region = 'SR'
+        if isDYCR:
+            region = 'DYCR'
+            channels='eE,eMu,muMu'
+        if isTTCR:
+            region = 'TTCR'
+            if isSC:
+                channels='eTau,muTau,tauTau'
+            else:
+                channels='eE,eMu,muMu'
         with input_file.localize("r") as local_input, self.output().localize("w") as local_output:
-            print(local_output.path)
-
             HistProducerFile_cmd = [ 'python3', HistProducerFile,
                                     '--inFile', local_input.path, '--outFileName',local_output.path,
                                     '--dataset', sample_name, '--uncConfig', unc_config,
                                     '--histConfig', self.setup.hist_config_path, '--sampleType', sample_type,
-                                    '--globalConfig', global_config, '--var', var, '--period', self.period  ]
-            if self.global_params['compute_unc_histograms'] or var == 'kinFit_m':
+                                    '--globalConfig', global_config, '--var', var, '--period', self.period, '--region', region, '--channels', channels]
+            # if self.global_params['compute_unc_histograms'] or var == 'kinFit_m':
+            if (var=='bbtautau_mass' or var == 'kinFit_m') and self.global_params['compute_unc_histograms'] and (isSC==True):
                 HistProducerFile_cmd.extend(['--compute_rel_weights', 'True', '--compute_unc_variations', 'True'])
-            if 'deepTau2p5' in self.version.split('_'):
-                print("deepTau2p5 in use")
+            if is2p5:
                 HistProducerFile_cmd.extend([ '--deepTauVersion', 'v2p5'])
             if need_cache:
                 anaCache_file = self.input()[1]
+                print(anaCache_file)
                 with anaCache_file.localize("r") as local_anacache:
                     HistProducerFile_cmd.extend(['--cacheFile', local_anacache.path])
                     ps_call(HistProducerFile_cmd, verbose=1)
             else:
                 ps_call(HistProducerFile_cmd, verbose=1)
 
+
+
 class HistProducerSampleTask(Task, HTCondorWorkflow, law.LocalWorkflow):
     max_runtime = copy_param(HTCondorWorkflow.max_runtime, 2.0)
     n_cpus = copy_param(HTCondorWorkflow.n_cpus, 1)
+    use_FileTask = luigi.BoolParameter(default=True)
```

Comment:
What is use-cases when fileTask is not needed? I propose to undo this modification

Reply:
If you remove the histograms from histProducerFileTask to save space!